jabgui/src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -98,4 +99,45 @@ void relinksMovedFile(@TempDir Path tempDir) throws IOException {
                 Path.of(matches.getFirst().getLink())
         );
     }
+
+    @Test
+    void findAllAssociatedNotLinkedFilesInsteadOfTheFirstOne(@TempDir Path tempDir) throws IOException {
+        Path directory = tempDir.resolve("files");
+        Path oldPath = directory.resolve("old/minimal.pdf");
+        Files.createDirectories(oldPath.getParent());
+        Files.createFile(oldPath);
+
+        LinkedFile stale = new LinkedFile("", oldPath.toString(), "PDF");
+        BibEntry entry = new BibEntry(StandardEntryType.Misc);
+        entry.addFile(stale);
+
+        // Simulate a file move
+        Path newPath1 = directory.resolve("new1/minimal.pdf");
+        Files.createDirectories(newPath1.getParent());
+        Files.move(oldPath, newPath1);
+
+        // Create a second copy of the file
+        Path newPath2 = directory.resolve("new2/minimal.pdf");
+        Files.createDirectories(newPath2.getParent());
+        Files.copy(newPath1, newPath2);
+
+        BibDatabaseContext context = mock(BibDatabaseContext.class);
+        when(context.getFileDirectories(filePreferences)).thenReturn(List.of(directory));
+
+        AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
+                context,
+                externalApplicationsPreferences,
+                filePreferences,
+                autoLinkPrefs);
+
+        List<LinkedFile> matches = util.findAssociatedNotLinkedFiles(entry);
+
+        assertEquals(2, matches.size());
+        assertTrue(matches.stream().anyMatch(file -> {
+            return FileUtil.relativize(newPath1, List.of(directory)).equals(Path.of(file.getLink()));
+        }));
+        assertTrue(matches.stream().anyMatch(file -> {
+            return FileUtil.relativize(newPath2, List.of(directory)).equals(Path.of(file.getLink()));
+        }));
```

Comment:
I think, `FileUtil` is used, but not scope of the test

Just test `matches` with a `List.of(...)` w/o calling `FileUtil`. I think, it should be easy to hard code the expected result?

Reply:
That makes sense.  Now the `FileUtil` is avoided.